### PR TITLE
flex porting

### PR DESCRIPTION
**Pull request by tdejoigny-ledger**
- flex porting
- switch to NBGL V2 (new use cases)
- add ticker for amounts

### CODE REVIEW

File: src/monero_ux_nbgl.c [hunk 1]
```diff
@@ -248,13 +255,13 @@ static void continue_display_last(int token, unsigned char index) {
...
-    transactionContext.infoLongPress.longPressText = "Approve";
+    transactionContext.infoLongPress.longPressText = "Hold to sign";
     transactionContext.infoLongPress.longPressToken = 0;
     transactionContext.infoLongPress.tuneId = TUNE_TAP_CASUAL;
-    transactionContext.infoLongPress.text = "Hold to confirm";
+    transactionContext.infoLongPress.text = "Sign transaction";
 
     nbgl_useCaseStaticReview(&transactionContext.tagValueList, &transactionContext.infoLongPress,
-                             "Cancel", ui_menu_validation_action);
+                             "Reject transaction", ui_menu_validation_action);
```
Comment (cedelavergne-ledger) [inline src/monero_ux_nbgl.c:264]: Isn't it possible to simplify and `use nbgl_useCaseReview` instead of `nbgl_useCaseStaticReview`?
It would simplify the configuration...

File: src/monero_prehash.c [hunk 2]
```diff
@@ -56,6 +56,9 @@ int monero_apdu_mlsag_prehash_init() {
         // fee str
         monero_vamount2str(G_monero_vstate.io_buffer + G_monero_vstate.io_offset,
                            G_monero_vstate.ux_amount, 15);
+
```
Comment (cedelavergne-ledger) [inline src/monero_prehash.c:59]: Just a question: why is this needed in the context of Flex Porting??

File: src/monero_ux_nbgl_menu.c [hunk 3]
```diff
@@ -117,13 +96,22 @@ static bool account_settings_navigation_cb(uint8_t page, nbgl_pageContent_t* con
         content->choicesList.names = accountNames;
         content->choicesList.token = ACCOUNT_CHOICE;
     } else if (page == 1) {
-        if (N_monero_pstate->account_id > 4) {
-            content->choicesList.initChoice = N_monero_pstate->account_id - 5;
+        if (N_monero_pstate->account_id > 3) {
```
Comment (cedelavergne-ledger) [inline src/monero_ux_nbgl_menu.c:99]: Does it mean the elements are hardcoded to be on page1 or page2?
Couldn't we have something more dynamique with nbglv2?

File: src/monero_ux_nbgl_menu.c [hunk 4]
```diff
@@ -83,32 +85,9 @@ static void update_account(void) {
...
-    } else {
-        return false;
-    }
-    return true;
-}
-
 static bool account_settings_navigation_cb(uint8_t page, nbgl_pageContent_t* content) {
     content->type = CHOICES_LIST;
-    content->choicesList.nbChoices = 5;
+    content->choicesList.nbChoices = 4;
```
Comment (cedelavergne-ledger) [inline src/monero_ux_nbgl_menu.c:90]: Why do we have 4 and no more 5? Is it linked to the split on 2 different pages?

File: tests/test_version.py [hunk 5]
```diff
@@ -8,7 +8,7 @@ def check_accepted_version(monero, valid_version: bytes):
     major, minor, patch = monero.reset_and_get_version(
         monero_client_version=valid_version
     )  # type: int, int, int
-    assert (major, minor, patch) == (1, 9, 0)  # version of the Monero app
+    assert (major, minor, patch) == (2, 0, 0)  # version of the Monero app
```
Comment (cedelavergne-ledger) [inline tests/test_version.py:11]: Could be usefull to get it directly from the Makefile: https://github.com/LedgerHQ/app-ethereum/blob/76e8690a68fb84910d09fbfd14cc975cefabe41a/tests/ragger/test_configuration_cmd.py#L9

File: tests/monero_client/monero_cmd.py [hunk 6]
```diff
@@ -474,8 +473,8 @@ def validate_prehash_update(self,
 
             navigator.navigate_and_compare(TESTS_ROOT_DIR,
```
Comment (cedelavergne-ledger) [inline tests/monero_client/monero_cmd.py:474]: I think you should replace the navigation by `scenario_navigator` functions, like https://github.com/LedgerHQ/app-ethereum/blob/76e8690a68fb84910d09fbfd14cc975cefabe41a/tests/ragger/test_domain_name.py#L70 ?

File: src/monero_ux_nbgl.c [hunk 7]
```diff
@@ -160,13 +160,13 @@ void ui_menu_change_validation_display_last(unsigned int value __attribute__((un
...
```
[the remark this answers is not in the document]
Comment (tdejoigny-ledger) [reply src/monero_ux_nbgl.c:169]: sadly this flow is split in several parts, so it's not possible to use the new `nbgl_useCaseReview `
refer to `ui_menu_validation_action`

File: src/monero_ux_nbgl.c (hunk 1 above)
Comment (tdejoigny-ledger) [reply src/monero_ux_nbgl.c:264]: sadly this flow is split in several parts, so it's not possible to use the new `nbgl_useCaseReview `
refer to `ui_menu_validation_action`

File: src/monero_prehash.c (hunk 2 above)
Comment (tdejoigny-ledger) [reply src/monero_prehash.c:59]: this change has been mentioned in the PR header. 
It's an UI improvement asked by @vforgeoux-ledger

File: src/monero_ux_nbgl_menu.c (hunk 3 above)
Comment (tdejoigny-ledger) [reply src/monero_ux_nbgl_menu.c:99]: That's what I initially tried to do, but unfortunately, it's not possible for multi-tiered settings.

File: src/monero_ux_nbgl_menu.c (hunk 4 above)
Comment (tdejoigny-ledger) [reply src/monero_ux_nbgl_menu.c:90]: because it does not fit on Flex

File: tests/test_version.py (hunk 5 above)
Comment (tdejoigny-ledger) [reply tests/test_version.py:11]: ok 👍

File: tests/monero_client/monero_crypto_cmd.py [hunk 8]
```diff
@@ -120,14 +120,13 @@ def display_address(self, test_name, firmware, navigator, derivation: bytes, out
...
-                NavIns(NavInsID.USE_CASE_CHOICE_CONFIRM),
-                NavIns(NavInsID.TOUCH, (200, 410)),
+                NavIns(NavInsID.SWIPE_CENTER_TO_LEFT),
+                NavIns(NavInsID.TOUCH, (200, 350 if firmware.device.startswith("flex") else 410)),
                 NavIns(NavInsID.USE_CASE_ADDRESS_CONFIRMATION_EXIT_QR),
-                NavIns(NavInsID.USE_CASE_ADDRESS_CONFIRMATION_TAP),
+                NavIns(NavInsID.SWIPE_CENTER_TO_LEFT),
                 NavIns(NavInsID.USE_CASE_ADDRESS_CONFIRMATION_CONFIRM),
                 NavIns(NavInsID.USE_CASE_STATUS_DISMISS)
             ]
```
[the remark this answers is not in the document]
Comment (tdejoigny-ledger) [reply tests/monero_client/monero_crypto_cmd.py:129]: cannot be used when the QR needs to be verified in the flow. It's interesting to check the QR code because Monero address is very long.

File: tests/monero_client/monero_cmd.py (hunk 6 above)
Comment (tdejoigny-ledger) [reply tests/monero_client/monero_cmd.py:474]: as you can see the flow for monero are not standard and scenario navigator can not be used easily.

**Review by cedelavergne-ledger — APPROVED** [review]
